lib/msal-common/src/cache/CacheManager.ts

```diff
@@ -661,7 +659,7 @@ export abstract class CacheManager implements ICacheManager {
      * @param homeAccountId
      */
     private matchHomeAccountId(entity: AccountEntity | CredentialEntity, homeAccountId: string): boolean {
-        return entity.homeAccountId && homeAccountId === entity.homeAccountId;
+        return !!(entity.homeAccountId && homeAccountId === entity.homeAccountId);
```

Comment:
Why are we adding a double boolean here?

Reply:
So I'm not sure I understand why but the compiler thinks the return type of the expression is `string | boolean`. I can't think of a scenario where `entity.homeAccountId` would be returned here since its an &&... but in any case prepending `!!` forces type `boolean`